src/lib/components/Map/map.tsx

```diff
@@ -0,0 +1,154 @@
+import React from "react";
+import { View } from "../View";
+import { Minimap } from "../Minimap";
+import { MapActions, MapActionType } from "../MapActions";
+import { Point } from "../../types/point";
+import { Size } from "../../types/size";
+import { useContainerDimensions } from "../../hooks/useContainerDimensions";
+import { useZoom } from "../../hooks/useZoom";
+import { usePrevious } from "../../hooks/usePrevious";
+import { DiagramConfig } from "lib/types/diagram";
+
+import "./map.css";
+
+type MapPropsType = {
+    Scene: React.ReactElement;
+    sceneSize: Size;
+    width: number | string;
+    height: number | string;
+    id: string;
+    config: DiagramConfig;
+};
+
+export const Map: React.FC<MapPropsType> = (props: MapPropsType): JSX.Element => {
+    const { sceneSize, Scene, width, height, id, config } = props;
+    const mapRef = React.useRef<HTMLDivElement>(null);
+    const size = useContainerDimensions(mapRef);
+    const [mapScale, setMapScale] = React.useState(1);
+    const [viewCenterPoint, setViewCenterPoint] = React.useState({ x: sceneSize.width / 2, y: sceneSize.height / 2 });
+    const [minimapCenterPoint, setMinimapCenterPoint] = React.useState({
+        x: sceneSize.width / 2,
+        y: sceneSize.height / 2,
+    });
```

Comment:
The problem is that we'd lose control over which component should be updated and which not. In the worst case, we might end up in an infinite loop.

Reply:
Nice, yes, both should update if center point changes, right?

Reply:
This would be possible when changing some of the code in `View` but it would result in a lot of rerenders of `View` (lacking a bit on my computer). Keeping it as is gives better performance. However, we could rename the state in `Map` to make it clearer?